src/Processors/Formats/Impl/ConstantExpressionTemplate.cpp

```diff
@@ -196,6 +203,13 @@ class ReplaceLiteralsVisitor
                 if (not_null == array.end())
                     return true;
             }
+            else if (literal->value.getType() == Field::Types::Map)
+            {
+                const Map & map = literal->value.get<Map>();
+                if (map.size() % 2)
```

Comment:
What does this condition mean? Why is map size odd if there are NULL values?

Reply:
But it not about null values. I think it's just an excessive check, which is actually unneeded.